File: Campus.Service.Address/Implementations/ServiceSettings.cs
using Campus.Service.Address.Interfaces;
using Campus.Service.Address.Interfaces.Settings;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Threading.Tasks;

namespace Campus.Service.Address.Implementations
{
    public class ServiceSettings : IServiceSettings
    {
        [JsonIgnore]
        private IResourceLoader resourceLoader = new ResourceLoader();
        public INetworkSettings networkSettings { get; set; } = new NetworkSettings();

        public IDatabaseSettings databaseSettings { get; set; } = new DatabaseSettings();

        public async Task intilizeServiceAsync()
        {
            string embeddedString = resourceLoader.GetEmbeddedResourceString(this.GetType().Assembly, "ServiceSettings.json");
            var temp = JsonConvert.DeserializeObject<ServiceSettings>(embeddedString);
            networkSettings = temp.networkSettings;
            databaseSettings = temp.databaseSettings;
            await networkSettings.intilizeSettingsAsync();
            await databaseSettings.intilizeSettingsAsync();
        }
    }
}

File: vueServer/vueserver/src/Custodial.Addressing.Service/ServiceSettings.ts
import { NetworkSettings } from './NetworkSettings';
import { DatabaseSettings } from './DatabaseSettings';
import { CasSetting } from './CasSettings';

export class ServiceSettings implements ServiceSettings{
    public networkSettings: NetworkSettings; 
    public databaseSettings: DatabaseSettings;
    public casSettings: CasSetting;

    constructor(networkSettings: NetworkSettings, databaseSettings: DatabaseSettings,
         casSettings: CasSetting){
        this.networkSettings = networkSettings;
        this.casSettings = casSettings;
        this.databaseSettings = databaseSettings;
    }


}